File: Day46/DSA/Find_Common_Elements_Between_Two_Arrays.cpp
Question:-  https://leetcode.com/problems/find-common-elements-between-two-arrays/

Code:-

class Solution {
public:
    vector<int> findIntersectionValues(vector<int>& nums1, vector<int>& nums2) {

        unordered_set<int> st1;
        unordered_set<int> st2;

        for (int i : nums1) {
            st1.insert(i);
        }

        for (int i : nums2) {
            st2.insert(i);
        }

        int answer1 = 0;
        int answer2 = 0;

        vector<int> ans;

        for (int i : nums1) {
            if (st2.find(i) != st2.end()) {
                answer1++;
            }
        }

        for (int i : nums2) {
            if (st1.find(i) != st1.end()) {
                answer2++;
            }
        }

        ans.emplace_back(answer1);
        ans.emplace_back(answer2);

        return ans;
    }
};

File: Day46/DSA/Find_Common_Elements_Between_Two_Arrays.java
Question:- https://leetcode.com/problems/find-common-elements-between-two-arrays/description/

Code:-  

class Solution {
    public int[] findIntersectionValues(int[] nums1, int[] nums2) {

        HashSet<Integer> set1 = new HashSet<>();
        HashSet<Integer> set2 = new HashSet<>();

        for(int i : nums1)
        {
            set1.add(i);
        }

        for(int i : nums2)
        {
            set2.add(i);
        }

        int answer1 = 0;
        int answer2 = 0;

        for(int i : nums1)
        {
            if(set2.contains(i))
            {
                answer1++;
            }
        }

        for(int i : nums2)
        {
            if(set1.contains(i))
            {
                answer2++;
            }
        }

        return new int[]{answer1,answer2};
    }
}